env/env.go

```diff
@@ -0,0 +1,32 @@
+package env
+
+import "os"
+
+// OsRepository ...
+type OsRepository interface {
```

Comment:
I think this name will be too ambiguous, even if used together with package name `env.OsRepository`.

I would much rather put this in the `os` package and name it `EnvironmentVariableRepository`: `os.EnvironmentVariableRepository`.
This way, the envman-based repository would read like this: `envman.EnvironmentVariableRepository`.

Wdyt?

Reply:
I agree. Also I think the term envar (not envvar) is in the lingo enough that we can use it instead of the bloated one: 
`os.EnvarRepository` `envman.EnvarRepository`, wdyt?

Reply:
I'm more in favor of unabbreviated, readable names (e.g. `EnvironmentVariableRepository`) because they are clear to everybody in any context (e.g. new joiners or members of other teams). But if others think it's in the Bitrise lingo then I'm okay with that too. 👍

Reply:
I assumed it is more than Bitrise lingo, like linux / generic OS lingo. `EnviromentVariable` seems painfully long if everybody knows what `Envar` is. I'm fine either way tho.

Reply:
After googling a bit, I'm slightly more convinced that `Envar` is a common word, but I'm still not entirely sure. 😄  Let's see what others think!

Reply:
I'm in favor of abbreviations where it's trivial to understand from the context, and I think this is the case here (Bitrise steps are all about env vars). But I don't like `Envar`, it seems like a typo to me 😅

Reply:
I also do not like `Envar` but suggest moving the os and envman env vars into different packages.

Reply:
Discussed offline, we settled with the following naming scheme:
* `os.EnvironmentRepository` for the OS-based environment variable storage.
* `envman.EnvironmentRepository` for the Envman-based environment variable storage.
* `Get/Set/ListVariable/s()` for function names.
* `defaultEnvironmentRepository` for the implementing `struct` (we are going to apply the same naming convention where interfaces and structs match 1:1).